settings: move ExternalIODir

RaduBerinde opened the pull request:
Move the `ExternalIODir` field from `cluster.Settings` to the server
config.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/138876)
<!-- Reviewable:end -->

RaduBerinde committed:
settings: move ExternalIODir

Move the `ExternalIODir` field from `cluster.Settings` to the server
config.

Epic: none
Release note: None

RaduBerinde commented:
Related - I think we should rename [ExternalIODirConfig](https://github.com/cockroachdb/cockroach/blob/ffa4eb13ef8999e9755e803e32692f8d200d433f/pkg/base/config.go#L1032) since its name makes no sense. I can do that in a separate PR. What would be a good name? `CloudServiceConfig`, `ExternalServiceConfig`?

RaduBerinde commented:
Friendly ping.

RaduBerinde commented:
TFTRs!

bors r+
